ff4s/src/main/scala/ff4s/Store.scala

```diff
@@ -34,53 +35,37 @@ trait Store[F[_], State, Action] {
 
 object Store {
 
-  def apply[F[_]: Concurrent, State, Action](
-      initialState: State
-  )(
-      makeDispatcher: SignallingRef[F, State] => Action => F[Unit]
+  def apply[F[_]: Concurrent, State, Action](init: State)(
+      update: Action => State => (State, F[Option[Action]])
```

Comment:
could it be `List[Action]`?

Reply:
Good point! And would you make them atomic, i.e., have a `Queue[F, List[Action]]`?

Reply:
hm, what do you mean by atomic in this context? e.g. what's an example of how `Queue[F, List[Action]]` would behave differently to `Queue[F, Action]`?

Reply:
Meaning that no other actions can be interspersed. Say my long-running effect returns `List(Inc, Inc)` and there is button dispatching `Reset`, then I could get the sequence `Inc, Reset, Inc` because I cannot guarantee atomic insertion of two elements into a queue.

Reply:
Ah indeed, you are right! Hmm 🤔 I think in that case the user should define a new type of action? Not sure, don't have a strong opinion. But it feels a bit inconsistent if some actions are atomic with each other depending on how they were submitted and other actions are atomic only to themselves.

In theory I suppose a user could even define a compound action that is just a list of actions and achieve the same semantic.

Reply:
Agree! I guess in that case keeping `F[Option[Action]]` is simpler and doesn't open up the question of atomicity. Or do you see any other benefits of `List` over `Option` that cannot be achieved with compound actions?

Reply:
Yes you're right, that could also be implemented in userland :)

Reply:
Perhaps it should be `Action => State => (State, Option[F[Option[Action]]])` for otherwise there is a call `supervisor.supervise(none.pure[F])` on every state update? 🤔

Reply:
I think I've convinced myself that this change is needed - for the sake of performance and to avoid having to write `none.pure[F]` or similar all over the place.